File: atcoder/abc106_b.cpp
// https://atcoder.jp/contests/abc106/tasks/abc106_b
#include<iostream>
//#include<algorithm>
//#include <numeric>
//#include<string>
//#include<vector>
//#include<map>
//#include<tuple>
//#include<queue>
//#include<regex>

// #include <bits/stdc++.h>
using namespace std;

int main() {
	int N;
	cin >> N;
	int ans = 0;
	int cnt;
	for (int i = 1; i < N+1; i += 2)
	{
		cnt = 0;
		for (int j = 1; j < i+1; j++)
		{
			if (i % j == 0)
			{
				cnt++;
			}
		}
		if (cnt == 8)
		{
			ans++;
		}
	}

	cout << ans << endl;
}

File: atcoder/abc106_b.py
# B - 105
# https://atcoder.jp/contests/abc106/tasks/abc106_b
# 105 : 1, 3, 5, 7, 15, 21, 35, 105

N = int(input())
ans = 0
for i in range(1, N+1, 2):
    cnt = 0
    for j in range(1, i+1):
        if i % j == 0:
            cnt += 1
    if cnt == 8:
        ans += 1

print(ans)
